### ISSUE TITLE
UprateMergeRequest optional fields

### ISSUE THREAD

**Issue by Gaff** [gitlab4j/gitlab4j-api#464]
Hi,

According to the Gitlab wiki, most of the arguments to `updateMergeRequest()` should be optional:
https://docs.gitlab.com/ee/api/merge_requests.html#update-mr

However the [API end point](http://www.messners.com/gitlab4j-api/javadocs/org/gitlab4j/api/MergeRequestApi.html#updateMergeRequest-java.lang.Object-java.lang.Integer-java.lang.String-java.lang.String-java.lang.Integer-java.lang.String-org.gitlab4j.api.Constants.StateEvent-java.lang.String-java.lang.Integer-java.lang.Boolean-java.lang.Boolean-java.lang.Boolean-java.lang.Boolean-) requires all of them to be present!

Would you be open to fixing this? I'm not quite sure how best to go about it. Ideas I can think of are:
 * Create extra updateXXX() methods - simple, but there are a LOT of permutations!
 * Rewrite the method to use `Optional<>` throughout - simple, albeit slightly unidomatic
 * Rewrite the method to be `updateMergeRequest(MergeRequest updatedObject)` - still not obvious which fields have changed. Would either need to perform a diff, or need some hibernate-like pojos that track what has changed.
* Use a builder-like pattern: 
```
   updateMergeRequestBuilder(p_id, mr_id)
        .setTitle(title)
        .setDescription(desc)
        .doUpdate();
```

Err - not sure where I'm going with this - have you got any preferences? If I added a MR that added a `updateMergeRequestDescription()` method, would you accept it? :)

**Comment by gmessner** [comment]
@Gaff 
All the parameters are not required to have a value, if you don't want the field modified simply pass ```null```.  That being said, we are addressing the **too many permutations** as changes are made in the various API classes and as requested.

This library has chosen a path similar to the builder pattern to address methods which require more than a few parameters.  What is done is a class is created to hold the parameters for the operation, the class would define ```withXXXX()``` methods to set the parameters for the operation.  For ```updateMergeRequest()``` it would look something like this:
```
updateMergeRequest(projectId, mergeRequestId,
    new MergeRequestParams().withTitle(title).withDescription(description))
```
I will be doing a release late 11/6/2019 and will add support for merge request in that release.

**Comment by Gaff** [comment]
Oh cool - that's excellent.  I shall look forward to tomorrow's patch! :)

If there's any work that needs doing I don't mind contributing some small MRs in this space.

**Comment by gmessner** [comment]
@Gaff 
I have made the updates described above and have released 4.12.18, please verify and close this issue when you have time.

**Comment by Gaff** [comment]
Looks great - thanks!
